components/list/src/lazy_automaton.rs

```diff
@@ -0,0 +1,79 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use regex_automata::dfa::sparse::DFA;
+use regex_automata::dfa::Automaton;
+use regex_automata::util::id::StateID;
+use writeable::Writeable;
+
+pub trait LazyAutomaton: Automaton {
+    // Like Automaton::find_earliest_fwd, but doesn't require a materialized string.
```

Comment:
Question: `find_earliest_fwd` says that it searches for the first match, whereas this function appears to require that the match appear at the beginning of the string; is that correct?

Reply:
The DFAs we construct are all "anchored", so only match at the beginning.

Reply:
See the test

Reply:
OK I think I get it now.